impl/maven-impl/src/test/java/org/apache/maven/impl/DefaultModelXmlFactoryTest.java

```diff
@@ -122,4 +125,65 @@ void testMalformedModelVersion() throws Exception {
         Model model = factory.read(request);
         assertEquals("invalid.version", model.getModelVersion());
     }
+
+    @Test
+    void testWriteWithoutFormatterDisablesLocationTracking() throws Exception {
+        // minimal valid model we can round-trip
+        String xml =
+                """
+                <project xmlns="http://maven.apache.org/POM/4.0.0">
+                  <modelVersion>4.0.0</modelVersion>
+                  <groupId>g</groupId>
+                  <artifactId>a</artifactId>
+                  <version>1</version>
+                </project>""";
+
+        Model model = factory.read(XmlReaderRequest.builder()
+                .reader(new StringReader(xml))
+                .strict(true)
+                .build());
+
+        StringWriter out = new StringWriter();
+        factory.write(XmlWriterRequest.<Model>builder()
+                .writer(out)
+                .content(model)
+                // no formatter -> tracking should be OFF
+                .build());
+
+        String result = out.toString();
+        // Heuristic: when tracking is off there should be no location output (no comments)
+        assertTrue(!result.contains("<!--"), "Expected no XML comments when location tracking is disabled");
+        // And certainly not our marker
+        assertTrue(!result.contains("LOC_MARK"), "Unexpected marker found in output");
```

Comment:
This isn't needed; or maybe keep this and get rid of the previous assert; if so, assertFalse